l10n_nl_xaf_auditfile_export/tests/test_l10n_nl_xaf_auditfile_export.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2018-2020 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import base64
+import os
+from io import BytesIO
+from zipfile import ZipFile
+
+from lxml import etree
+
+from odoo import fields
+from odoo.tests.common import Form, TransactionCase
+from odoo.tools import mute_logger
+
+
+def get_transaction_line_count_from_xml(auditfile):
+    """Helper XML method to parse and return the transaction line count"""
+    line_count = 0
+    with ZipFile(BytesIO(base64.b64decode(auditfile)), "r") as z:
+        contents = z.read(z.filelist[-1]).decode()
+        parser = etree.XMLParser(
+            ns_clean=True, recover=True, encoding="utf-8", remove_blank_text=True
+        )
+        root = etree.XML(bytes(contents, encoding="utf8"), parser=parser)
+        # xpath query to select all element nodes in namespace
+        # Source: https://stackoverflow.com/a/30233635
+        query = "descendant-or-self::*[namespace-uri()!='']"
+        for element in root.xpath(query):
+            element.tag = etree.QName(element).localname
+        journals = root.xpath("/auditfile/company/transactions/journal")
+        for journal in journals:
+            transactions = journal.xpath("transaction/trLine")
+            for _ in transactions:
+                line_count += 1
+    return line_count
+
+
+class TestXafAuditfileExport(TransactionCase):
+    def setUp(self):
+        super().setUp()
+
+        self.env.user.company_id.country_id = (self.env.ref("base.nl").id,)
+
+        # create an invoice and post it, to ensure that there's some data to export
+        move_form = Form(
+            self.env["account.move"].with_context(default_move_type="out_invoice")
+        )
+        move_form.invoice_date = fields.Date().today()
+        move_form.partner_id = self.env["res.partner"].create({"name": "Partner Test"})
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.env["product.product"].create(
+                {"name": "product test", "standard_price": 800.0}
+            )
+        self.invoice = move_form.save()
+        self.invoice.post()
```

Comment:
There is a warning on this method.
DeprecationWarning: RedirectWarning method 'post()' is a deprecated alias to 'action_post()' or _post()